src/loaders/genome_collection/parse_biolog_data.py

```diff
@@ -0,0 +1,187 @@
+    # link to the file: https://docs.google.com/spreadsheets/d/1QmC6UHWOEVfpmrveRBl_izictbNeN1PA/edit#gid=1135979967
+
+    # Drop unused columns
+    data_df = data_df.drop(columns=['# pos. strains'])
+
+    # Make the first row ('strain designation') the column names
+    data_df.columns = data_df.loc[1]
+
+    # Drop unused rows ('total cpds. Utilized', 'strain designation' and unused rows at the end.)
+    data_df = data_df.drop(index=[0, 1, 194, 195, 196, 197, 198])
+
+    # Make the first column ('strain designation') the index
+    data_df = data_df.set_index('strain designation')
+
+    # Transpose the DataFrame
+    data_df.index.name = None
+    data_df.columns.name = None
+    data_df = data_df.transpose()
+
+    # Remove asterisks from index names
+    data_df.index = data_df.index.str.replace(r'*', '').str.replace(r'**', '')
+
+    # Merge rows BT03 A(10/1) and BT03 B(11/18) into one row BT03
+    data_df = data_df.drop(index='BT03 B(11/18)')
+    data_df = data_df.rename(index={'BT03 A(10/1)': 'BT03'})
+
+    # Replace NaN with 0
+    # TODO: this is a temporary solution to handle the missing data for some strains.
+    # Two NaN values are present in the data file for 'YR530' and 'CF286' in the row labeled '3-Hydroxy 2-Butanone'.
+    data_df.fillna(0, inplace=True)
+
+    return data_df
+
+
+def _read_biolog_meta(biolog_meta_file: Path) -> pd.DataFrame:
+    meta_df = _read_excel_as_df(biolog_meta_file)
+
+    return meta_df
+
+
+def _retrieve_kbase_genome_id(meta_df, strain_id) -> str:
+    # Retrieve the KBase genome ID for the given strain ID from the metadata file (dataframe).
+
+    matching_row = meta_df[(meta_df['Strain ID'] == strain_id) & meta_df['Genome_Ref'].notna()]
+
+    if matching_row.empty:
+        raise ValueError(f"No matching Strain ID with a non-NaN Genome_Ref found for {strain_id}.")
+    else:
+        genome_ref = matching_row['Genome_Ref'].values[0]
+
+        return genome_ref
+
+
+def generate_biolog_heatmap_data(
+        biolog_data_file: Path,
+        biolog_meta_file: Path,
+        load_ver: str,
+        env: str = 'PROD',
+        kbase_collection: str = 'biolog',
+        root_dir: str = loader_common_names.ROOT_DIR):
+    """
+    Generate JSONL files with the heatmap data for the Biolog data file.
+
+    :param biolog_data_file: Biolog data file
+    :param biolog_meta_file: Biolog meta file
+    :param load_ver: the version of the data to be loaded
+    :param env: the environment containing the data to be processed
+    :param kbase_collection: the name of the KBase collection
+    :param root_dir: the root directory for the data to be processed
+    """
+    data_df = _read_biolog_data(biolog_data_file)
+    meta_df = _read_biolog_meta(biolog_meta_file)
+
+    heatmap_cell_details, heatmap_rows, heatmap_meta_dict = list(), list(), dict()
+
+    # create document for heatmap meta
+    categories = list()
+    for idx, metabolite in enumerate(data_df.columns):
```

Comment:
A lot of this code looks very similar to the parser code. Should it be moved to a shared module?

I was originally thinking this should just output data as if it were run by a tool container and have the parser summarize it but that might be overkill.

Reply:
Yes, I considered that approach. However, for the microtrait dataset, we had separate folders with data for each row, which required processing and merging. In contrast, for this dataset, we already have all the necessary data, so I built all the models simultaneously. While there are similarities, the two approaches are also quite distinct.

Reply:
Ok, my main concern here is that if the heatmap data structures change we'll need to update 2 places and we might forget. Can you add comments to this file and the parser pointing out the similar code in the other file and the need to update both if they change?

Reply:
👍

Reply:
:+1: